### PR TITLE
Validate facilitator config contents

### LINKED ISSUE #9: Config:  Validate facilitator-config contents
This is an implementation ticket. The actual implementation may vary as we start working on it.
Please check #2 first for any updates in params.

The `facilitator-config` file provided must be verified before we start the facilitator.

Implement the function to validate the facilitator-config.

- Check #2 for the interface of the function.
- Define the facilitator-config schema.
- Validate if facilitator-config against the schema. <strike> contains all the mandatory params required for facilitation.
    - RPC endpoint for the chains
    - worker addresses.
- Validate DB configs, if it exists or not.</strike>
- use npm tool to match against schema

### PR DESCRIPTION
**Pull request by gulshanvasnani**
Merge after https://github.com/OpenST/facilitator/pull/65
PR contains changes to validate facilitator config. It validates the config against the json schema.

fixes #9

### CODE REVIEW

**Comment by gulshanvasnani** [comment]
Will resume it after PR #65  gets merged.

**Comment by abhayks1** [comment]
Started reviewing 🐮 🐃

**Comment by abhayks1** [comment]
Unassigning myself as didn't get much time to review today

File: src/Config.ts [hunk 1]
```diff
@@ -105,11 +108,26 @@ export class FacilitatorConfig {
...
     }
     return new FacilitatorConfig({});
   }
 
+  /**
+   * This method validate json object against facilitator config schema and throws
+   * an exception on failure.
+   * @param jsonObject JSON object to be validated against schema.
+   */
+  public static validateSchema(jsonObject: any): void {
```
Comment (deepesh-kn) [inline src/Config.ts]: rename it to `verifySchema`

File: src/Config.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import * as path from 'path';
 import * as fs from 'fs-extra';
 import { EncryptedKeystoreV3Json } from 'web3-eth-accounts';
+import { Validator } from 'jsonschema';
```
Comment (deepesh-kn) [inline src/Config.ts]: import this as `JsonSchemaVerifier`

File: test/Facilitator/from.test.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+    assert.strictEqual(
+      facilitatorConfig.database.path,
+      undefined,
+      'Facilitator DB path is incorrect'
+    );
+
+    sinon.restore();
+  });
+});
```
Comment (deepesh-kn) [inline test/Facilitator/from.test.ts]: Add more test case  
`FacilitatorConfig.validateSchema` throws  `InvalidFacilitatorConfigException`  exception.

File: test/FacilitatorConfig/from.test.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import Utils from '../../src/Utils';
+import SpyAssert from "../utils/SpyAssert";
+import { FacilitatorConfig } from "../../src/Config";
+import {assert} from 'chai';
```
Comment (abhayks1) [inline test/FacilitatorConfig/from.test.ts]:
```suggestion
import { assert } from 'chai';
```

File: test/FacilitatorConfig/constructor.test.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import {assert} from 'chai';
+import Utils from '../../src/Utils';
+import * as path from 'path';
+import {FacilitatorConfig} from '../../src/Config';
```
Comment (abhayks1) [inline test/FacilitatorConfig/constructor.test.ts]: Please run linter for all the test files.

```suggestion
import {FacilitatorConfig} from '../../src/Config';
```
```suggestion
import { FacilitatorConfig } from '../../src/Config';
```

File: test/FacilitatorConfig/from.test.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+const sinon = require('sinon');
```
Comment (abhayks1) [inline test/FacilitatorConfig/from.test.ts]: This gives linting error:
```
import * as sinon from 'sinon';
```

File: src/Exception.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Defines error for facilitator config.
+ */
+export default class InvalidFacilitatorConfigException extends Error {
+  public constructor(message: string) {
```
Comment (abhayks1) [inline src/Exception.ts]: Documentation missing 🎶

**Pull request merged**